python/revng/scripting/local_daemon_profile.py

```diff
@@ -0,0 +1,91 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import socket
+import subprocess
+from signal import SIGINT
+from time import sleep
+
+import requests
+
+from revng.scripting.projects import DaemonProject
+
+
+class LocalDaemonProject(DaemonProject):
+    """
+    This class extends the DaemonProject. When initialized it starts the revng daemon
+    server and setups the DaemonProject client used to connect to the server.
+    """
+
+    def __init__(self, resume_dir: str | None = None, revng_executable: str | None = None):
+        self.port: int = self._get_port()
+        self.daemon_process: subprocess.Popen | None = None
+        self.resume_dir = self._set_resume_dir(resume_dir)
+        self.revng_executable = self._set_revng_executable(revng_executable)
+        self.start_daemon()
+        super().__init__(f"http://127.0.0.1:{self.port}")
+
+    def __del__(self):
+        self.stop_daemon()
+
+    def start_daemon(self, connection_retries: int = 10):
+        """
+        Start the `revng` daemon and wait for it to be ready.
+        """
+        env = os.environ
+        env["REVNG_DATA_DIR"] = self.resume_dir
+
+        cli_args = [self.revng_executable, "daemon", "-b", f"tcp:127.0.0.1:{self.port}"]
+        self.daemon_process = subprocess.Popen(
+            cli_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT, env=env
+        )
+
+        failed_retries = 0
+        while failed_retries <= connection_retries:
+            if self._is_server_running():
+                return
+            failed_retries += 1
+            sleep(1)
+        raise RuntimeError(f"Couldn't connect to daemon server at http://127.0.0.1:{self.port}")
+
+    def stop_daemon(self) -> int:
+        """
+        Stop the daemon server.
+        """
+        if self.daemon_process:
+            self.daemon_process.send_signal(SIGINT)
+            return self.daemon_process.wait(30.0)
```

Comment:
Add `self.daemon_process = None` somewhere here so that `stop_daemon` is re-entrant